voting/src/general_meeting.rs

```diff
@@ -119,7 +136,56 @@ impl Context {
 
                 Ok(())
             }
+            PUBLISH_RESULT_TX_TYPE => {
+                let tx: PublishResultTransaction =
+                    serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+
+                let meeting_id = tx.action.meeting_id;
+                let key = crate::generate_voting_box_key(&meeting_id);
+                let vote_box: VoteBox = {
+                    let bytes = &self.storage().get(key.as_slice()).ok_or_else(|| ExecuteError::VotingBoxNotFound)?;
+                    serde_cbor::from_slice(&bytes).expect("Vote box is serialized by this code")
+                };
 
+                let mut meeting: GeneralMeeting = {
+                    let bytes =
+                        self.storage().get(meeting_id.as_ref()).ok_or_else(|| ExecuteError::GeneralMeetingNotFound)?;
+                    serde_cbor::from_slice(&bytes).expect("General meeting is serialized by this code")
+                };
+
+                let now = self.block_header.as_ref().unwrap().timestamp();
+                if meeting.tallying_time.time > now {
+                    return Err(ExecuteError::PublishTimeBeforeTallyingTime)
+                }
+                let mut final_result = Vec::new();
+
+                for agenda in 0..meeting.number_of_agendas {
+                    let mut vote_result = VoteResult::new();
+                    for vote_id in &vote_box.votes {
+                        let vote = self.vote().get_vote(&vote_id).unwrap();
+                        let vote_choice = vote.get_choice();
+                        let number_of_shares = self.vote().get_shares(&vote_id).unwrap();
```

Comment:
Isn't it enough to use `shares`?
```suggestion
                        let shares = self.vote().get_shares(&vote_id).unwrap();
```

Naming is a difficult work. If you think other name is better, please telle me.